bin/check-swap-percent.rb

```diff
@@ -44,6 +44,12 @@ def run
     total_swap = output_swap.split(' ')[1].to_i
     used_swap = output_swap.split(' ')[2].to_i
 
+    # Check if swap exists
+    if total_swap.zero?
+      print 'There is no SWAP on this machine'
+      exit
```

Comment:
Swap is optional unless it is a guided installation and not uncommon way to setup a system. Probably more popular with VM's. I don't see any point to warn/crit about swap missing on systems that were setup like that in the very beginning. Or, maybe I misunderstood your question?